snuba/cli/optimize.py

```diff
@@ -3,26 +3,30 @@
 
 import click
 
-from snuba import settings
+from snuba import settings, datasets
+from snuba.datasets.factory import get_dataset
 
 
 @click.command()
 @click.option('--clickhouse-server', multiple=True,
               help='Clickhouse server to optimize.')
 @click.option('--database', default='default',
               help='Name of the database to target.')
-@click.option('--table', default=settings.DEFAULT_LOCAL_TABLE,
-              help='Name of the table to target.')
+@click.option('--dataset', default='events', type=click.Choice(['events']),
+              help='The dataset to target')
 @click.option('--timeout', default=10000, type=int,
               help='Clickhouse connection send/receive timeout, must be long enough for OPTIMIZE to complete.')
 @click.option('--log-level', default=settings.LOG_LEVEL, help='Logging level to use.')
-def optimize(clickhouse_server, database, table, timeout, log_level):
+def optimize(clickhouse_server, database, dataset, timeout, log_level):
     from datetime import datetime
     from snuba.clickhouse import ClickhousePool
     from snuba.optimize import run_optimize, logger
 
     logging.basicConfig(level=getattr(logging, log_level.upper()), format='%(asctime)s %(message)s')
 
+    dataset = get_dataset(dataset)
+    table = dataset.SCHEMA.get_table_name()
```

Comment:
Shouldn't this explicitly pin to the local table?

Reply:
fixed